### DISCUSSION TITLE
Is it possible to nest Semantic tokens

### DISCUSSION THREAD

**Discussion by brandenTenbrink** [chakra-ui/chakra-ui#7453] [Q&A] [answered]
Perhaps I am implementing this wrong for my component library, but I noticed that the semantic tokens only seem to work for 1 nested layer under colors, so

```
semanticTokens: {
  colors: {
    disabled: { ... }
```

would work when I do color: 'disabled', but if I try to make the object any more complex, such as:

```
semanticTokens: {
  colors: {
    text: {
      purple: {
        pressed: { ... }
```

Calling it via color: 'text.purple.pressed' no longer works.

I am trying to implement a semantic system that is capable of support something like this: <img width="714" alt="Screenshot 2023-03-14 at 2 05 24 PM" src="https://user-images.githubusercontent.com/15152523/225097600-84d01773-ae96-4375-98da-a923f0587ede.png">

Is this something that is currently possible with Chakra? And also is this even the correct way to be handling multiple potential colorSchemes. By declaring each color, I figured the colors could essentially become a prop via the component styles like such:

```
    baseStyle: defineStyle((props) => {
      const { colorScheme } = props;
      return {
        color: `text[${colorScheme}][pressed]`
```

**Comment by kuroppe1819** [comment, marked as answer]
I gave it a try implementation to declare nesting object in the semantic tokens.
https://github.com/chakra-ui/chakra-ui/pull/7494

**Reply by segunadebayo** [reply]
Good suggestion and I love how you followed it up with a PR.

**Reply by kuroppe1819** [reply]
Thank you for your review and for merging!
